js/main.js

```diff
@@ -81,21 +109,56 @@ function renderEntriesLoading(event) {
 
 window.addEventListener('DOMContentLoaded', renderEntriesLoading);
 
+function chooseToEdit(event) {
+  if (event.target && event.target.matches('i')) {
+    getEntry = event.target.closest('[data-entry-id]');
+    $liList = document.querySelectorAll('[data-entry-id]');
+    for (var i = 0; i < $liList.length; i++) {
+      if ($liList[i] === getEntry) {
+        var titleEdit = data.entries[i].title;
+        var photoUrlEdit = data.entries[i].photoUrl;
+        var notesEdit = data.entries[i].notes;
+        data.editing = data.entries[i].EntryId;
+        $createEntry.elements.title.value = titleEdit;
+        $createEntry.elements.photoUrl.value = photoUrlEdit;
+        $createEntry.elements.notes.value = notesEdit;
+        $img.src = photoUrlEdit;
+      }
+    }
+  }
+}
+$ul.addEventListener('click', chooseToEdit);
+$createEntry.addEventListener('click', chooseToEdit);
+
 function viewChange(event) {
   if (event.target && event.target.matches('.href-new-entries')) {
     $entryForm.className = '';
     $entries.className = 'hidden';
+    $editH1.className = 'edit-entry hidden';
+    $newH1.className = 'new-entry ';
     data.view = 'entry-form';
   }
 
   if (event.target && event.target.matches('.href-entries-list')) {
     $entries.className = '';
     $entryForm.className = 'hidden';
+    $editH1.className = 'edit-entry hidden';
+    $newH1.className = 'new-entry ';
+    $createEntry.reset();
+    $img.src = 'images/placeholder-image-square.jpg';
     data.view = 'entries';
   }
 
+  if (event.target && event.target.matches('i')) {
+    $entries.className = 'hidden';
+    $entryForm.className = '';
+    $editH1.className = 'edit-entry ';
+    $newH1.className = 'new-entry hidden';
+    data.view = 'edit';
+  }
```

Comment:
This setup is **_very_** manual. If you want to add a new view, you will need to revise this function every time you do, forever. Instead, you should query for all view container, and loop over that list. For each view in the list, check if that view is the view that is meant to be shown, and if it is, show it. Otherwise, hide it. I suggest you review `javascript-view-swapping` again for a reminder of how to do this effectively.